### ISSUE TITLE
Auto Updating doesn't seem to work

### ISSUE THREAD

**Issue by marcelharinck** [XInTheDark/raycast-g4f#113]
**Description**
Not able to update from 3.4 to 3.5

**To Reproduce**
1. Go to Check for Updates
2. Click on 'Update' when asked
4. Immediately see error 'Update Failed'

**Expected behavior**

**Screenshots/Recordings**
![image](https://github.com/user-attachments/assets/f67f68b2-cc8d-470a-b26a-57da57a8794a)
![image](https://github.com/user-attachments/assets/5a7ee609-cd04-46fb-a5b5-854119774982)

**Information**
 - raycast-g4f version: v3.4
 - Raycast version: v1.82.0

**Comment by XInTheDark** [comment]
Hi! In the source code folder, can you: 
- run `npm run dev` in the terminal
- then run the Check For Updates command again
and share the output printed? Thanks.

**Comment by marcelharinck** [comment]
There you go:

```
Error: Update failed

update.jsx:80:12

---
77:       popToRoot();
78:     } else {
79:       showToast(Toast.Style.Failure, "Update failed");
80:       throw new Error("Update failed");
81:     }
82:   });
83: };
---

ChildProcess.exithandler:node:child_process:430:5
ChildProcess.emit:node:events:519:28
ChildProcess.emit:node:domain:488:12
```

**Comment by XInTheDark** [comment]
Hey! Sorry for the delay. Could you list the output of the following commands in terminal:
- git -v
- npm -v
- which npm
- where npm
- curl -V

thanks!

**Comment by marcelharinck** [comment]
Hmm seem to have both though:

`git version 2.46.0
10.8.2
/opt/homebrew/bin/npm
/opt/homebrew/bin/npm
curl 8.7.1 (x86_64-apple-darwin23.0) libcurl/8.7.1 (SecureTransport) LibreSSL/3.3.6 zlib/1.2.12 nghttp2/1.61.0
Release-Date: 2024-03-27
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns ldap ldaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS GSS-API HSTS HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM SPNEGO SSL threadsafe UnixSockets
mharinck@NL-H322HVW35J ~ %`

**Comment by XInTheDark** [comment]
Hey, I just resolved this in eddf102; it will also be available in the next update. Could you confirm that it works after the next update? Thanks.

**Comment by XInTheDark** [comment]
Hi @aloshev! I'll look more into the auto-update

Could you check that the latest version of the update script is still not working? (e.g. by manually decreasing the version in package.json then running the update command). In particular I've recently updated this: https://github.com/XInTheDark/raycast-g4f/blob/1360e6871a355531d4642b45b046350c4372168c/assets/scripts/update.sh#L11

which should probably fix the Homebrew path issue.

Regarding the second point about using `npm run build` - does the extension simply not update when you run it manually on the command line? If so, I might need to ask the raycast team, and possibly switch to using `npm run dev` everywhere.

Thanks!

**Comment by XInTheDark** [comment]
Hey - I can confirm that this happens.

After doing a bit of experimenting, apparently:
- `npm run build` updates the js code of the extension commands, but doesn't update the extension preferences and related info. e.g. I can see that the commands were updated (and btw I only chceked for this when introducing the auto-update functionality, so the preferences bug went unnoticed). This is likely a bug on Raycast's side, since the docs/instructions given were that `npm run build` is sufficient to update the extension.
- `npm run dev` works fine to update the entire extension.

Since this is the case, I'll be updating all the scripts and instructions to use only `npm run dev`. Will get back to you soon!

**Comment by XInTheDark** [comment]
I've hopefully fixed this in #123

**Comment by XInTheDark** [comment]
Hi, regarding this:

> Does not exit developer mode after killing npm
> I suggest killing the ray process instead - kill $(pgrep ray)

Is there a specific reason for this? Not sure what the `ray` process does, beside building the extension then communicating with Raycast app. Killing the developer mode historically worked fine, since there wouldn't be any difference to the extension after developer mode is killed

**Comment by XInTheDark** [comment]
Ohh I see - does terminating the ray process work for you? If so I'll push a bug fix

**Comment by XInTheDark** [comment]
Done with 3f008bf2c17faa65674969f2b0acdcfe0cb6a8ec - can you check if it works for you now?

**Comment by XInTheDark** [comment]
fixes released in v3.9.1, will close this issue for now :)
